### PR TITLE
Sort lookup tables in vellum

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?186389

cc; @TylerSheffels

### CODE REVIEW

File: corehq/apps/app_manager/views/formdesigner.py [hunk 1]
```diff
@@ -189,7 +189,10 @@ def get_data_schema(request, domain, app_id=None, form_unique_id=None):
     if app.domain != domain:
         raise Http404()
     data.append(get_casedb_schema(app))  # TODO use domain instead of app
-    data.extend(item_lists_by_domain(domain))
+    data.extend(
+        sorted(item_lists_by_domain(domain),
+               cmp=lambda x,y: cmp(x['name'].lower(), y['name'].lower()))
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/formdesigner.py]: E231 missing whitespace after ','

File: corehq/apps/app_manager/views/formdesigner.py [hunk 2]
```diff
@@ -189,7 +189,10 @@ def get_data_schema(request, domain, app_id=None, form_unique_id=None):
     if app.domain != domain:
         raise Http404()
     data.append(get_casedb_schema(app))  # TODO use domain instead of app
-    data.extend(item_lists_by_domain(domain))
+    data.extend(
+        sorted(item_lists_by_domain(domain),
+               cmp=lambda x, y: cmp(x['name'].lower(), y['name'].lower()))
```
Comment (millerdev) [inline corehq/apps/app_manager/views/formdesigner.py]: Why not `key=lambda x: x['name'].lower()`? I think `cmp=...` is deprecated.

Comment (emord) [inline corehq/apps/app_manager/views/formdesigner.py]: Changed. It doesn't mention being deprecated in python 2 docs, but it doesn't exist in 3.5

Comment (millerdev) [inline corehq/apps/app_manager/views/formdesigner.py]: You're right, not deprecated. It's just ["the old way."](https://wiki.python.org/moin/HowTo/Sorting#The_Old_Way_Using_the_cmp_Parameter)

**Comment by millerdev** [comment]
GTM when tests pass (are there any tests that might care about this change?)

**Comment by emord** [comment]
I don't think there is, but it's not urgent so no need to merge

**Comment by czue** [comment]
nice, thanks!

**Pull request merged**